surefire-provider/src/test/java/org/arquillian/smart/testing/surefire/provider/SmartTestingProviderTest.java

```diff
@@ -101,6 +113,17 @@ public void test_when_invoke_is_called_with_whole_set_of_classes() throws Except
             iterable -> iterableContains(iterable, expectedClassesToRun)));
     }
 
+    @BeforeClass
+    public static void dumpConfiguration() {
+        Configuration.load().dump();
+    }
+
+    @AfterClass
+    public static void purgeConfiguration() {
+        final Path tempDirPath = Paths.get(Paths.get("").toAbsolutePath().toString(), SMART_TESTING_WORKING_DIRECTORY_NAME);
+        FileSystemOperations.deleteDirectory(tempDirPath, true);
```

Comment:
wouldn't be better to use `TemporaryFolder` rule?

Reply:
We can't use it `TemporaryFolder` Rule because it has it's own format for naming `given_name_time_stamp`. We need to dump configuration to `${projectdir}/.smart-testing/temporary/smart-testing.yml` So `Configuration.loadPrecalculated(File projectDir)` method can load configuration from that predefined location.

Reply:
setting the path to the project root directory should solve this problem - see my comment: https://github.com/arquillian/smart-testing/pull/183#discussion_r140525783